File: core/src/main/java/com/fahamutech/kcore/admin/database/connector/CategoryDataSource.java
package com.fahamutech.kcore.admin.database.connector;

import com.fahamutech.kcore.admin.database.DataBaseCallback;
import com.fahamutech.kcore.admin.model.ICategory;

public interface CategoryDataSource {
    void createCategory(ICategory category, DataBaseCallback... callbacks);

    void deleteCategory(String docId, DataBaseCallback... callbacks);
}

File: core/src/main/java/com/fahamutech/kcore/admin/database/connector/CategoryDataSource.kt
package com.fahamutech.kcore.admin.database.connector

import com.fahamutech.kcore.admin.database.DataBaseCallback
import com.fahamutech.kcore.admin.model.Category

interface CategoryDataSource {
    fun createCategory(category: Category?, vararg callbacks: DataBaseCallback?)
    fun deleteCategory(docId: String?, vararg callbacks: DataBaseCallback?)
}